src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2021 Hunter Monroe. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hunter Monroe, Kyle Miller, Alena Gusakov
+-/
+
+import combinatorics.simple_graph.basic
+
+/-!
+# Subgraphs of a simple graph
+
+This file formalizes a subgraph of a simple graph as a structure and proves a few simple lemmas.
+-/
+
+universe u
+variables {V : Type u} (G : simple_graph V)
+open simple_graph
+
+/-- A subgraph of a graph is a subset of vertices and a subset of edges such that each endpoint of
+an edge is contained in the vertex set. Subgraphs implement the `simple_graph` class. -/
+@[ext]
+structure subgraph :=
+(V' : set V)
+(adj' : V → V → Prop)
+(adj_sub : ∀ ⦃v w : V⦄, adj' v w → G.adj v w)
+(edge_vert : ∀ ⦃v w : V⦄, adj' v w → v ∈ V')
+(sym' : symmetric adj')
+
+/-- The Prop that states that `G'` is isomorphic to a subgraph of `G`. -/
+def is_subgraph (G' : simple_graph V) : Prop := ∀ ⦃v w : V⦄, G'.adj v w → G.adj v w
+
+namespace subgraph
+
+variable {G}
+
+/-- The edges of `G'` consist of a subset of edges of `G` -/
```

Comment:
```suggestion
/-- The edges of `G'` consist of a subset of edges of `G`. -/
```
I'm also not sure about the convention for `'` in this file.  I think in one version of it, I had removed them all since it doesn't add that much -- it makes dot notation cleaner without, though you have to write `subgraph.mem_edge_set` occasionally, for example.

Reply:
Not sure I understand what you are suggesting.

Reply:
I'm not sure whether lemmas that are subgraph-specific should have a `'` in the name.  I'm curious what others think.

Maybe the reason I added the primes is that `simple_graph.subgraph.mem_edge_set` overrides `simple_graph.mem_edge_set` in this context.  Maybe it's worth making the `simple_graph` in each of these explicit arguments so you can write `G.mem_edge_set` to disambiguate.